be/src/olap/tablet_reader.cpp

```diff
@@ -276,6 +276,8 @@ TabletColumn TabletReader::materialize_column(const TabletColumn& orig) {
                                cast_type.type);
     }
     column_with_cast_type.set_type(filed_type);
+    column_with_cast_type.set_precision_frac(cast_type.precision, cast_type.scale);
+    column_with_cast_type.set_is_decimal(cast_type.precision > 0);
```

Comment:
It's not rigorous to use `cast_type.precision > 0` to judge is_decimal. You can use cast_type.is_decimal_v2_type() and cast_type.is_decimal_v3_type() or add a new is_decimal_type for cast_type.